### PR TITLE
[WIP] Test cases for relations filters

### PR DESCRIPTION
**Pull request by gargol**
closes https://github.com/NexesJS/mongo-knex/issues/3

- Added base cases for M:M relations filtering

Needs more discussion around the format of relationship description configuration and kinds or relations that need to be tested

### CODE REVIEW

File: test/integration/relations.test.js [hunk 1]
```diff
@@ -164,4 +164,247 @@ describe.skip('Relations', function () {
...
+                        // Check we get the right data
+                        // result.should.do.something;
+                    });
+            });
+        });
+
+        // NOTE: only $lt and $gt operators are supported by nql-lang, no need to support $eq operator for now
+        describe('count', function () {
+            it('can compare by count $gt', function () {
+                const queryJSON = {'authors.posts': {$gt: 0}};
```
Comment (kirrg001) [inline test/integration/relations.test.js]: IMO this must be count.posts?

File: test/integration/relations.test.js [hunk 2]
```diff
@@ -164,4 +164,247 @@ describe.skip('Relations', function () {
...
+                        //       do not count in posts with multiple tags containing one of the excluded tags
+                        result.should.be.an.Array().with.lengthOf(3);
+
+                        // Check we get the right data
+                        // result.should.do.something;
+                    });
+            });
+
+            it('can match array $nin (multiple values)', function () {
+                const queryJSON = {'tags.slug': {$nin: ['animal', 'classic']}};
```
Comment (kirrg001) [inline test/integration/relations.test.js]: have you covered aliases?

tag:en

**Comment by kirrg001** [comment]
I just left two tiny comments. If resolved, we should just merge this in 👍

File: test/integration/suite1/fixtures/many-to-many-extended-cases.json [hunk 3]
```diff
@@ -0,0 +1,105 @@
+{
```
Comment (kirrg001) [inline test/integration/suite1/fixtures/many-to-many-extended-cases.json]: base.json equals this file except of the relations or?

each suite has a base. and then a test can insert more data.

Comment (kirrg001) [reply test/integration/suite1/fixtures/many-to-many-extended-cases.json]: oh btw: let me know if you think the concept does not make sense to you :)

File: test/integration/relations.test.js (hunk 2 above)
Comment (gargol) [reply test/integration/relations.test.js]: good point adding them.

File: test/integration/relations.test.js (hunk 1 above)
Comment (gargol) [reply test/integration/relations.test.js]: the base model to `makeQuery` atm is made with `posts`, so this becomes `count.authors` I think

Comment (kirrg001) [reply test/integration/relations.test.js]: exactly, i put the the comment on the wrong use case 👍

File: test/integration/suite1/fixtures/many-to-many-extended-cases.json (hunk 3 above)
Comment (gargol) [reply test/integration/suite1/fixtures/many-to-many-extended-cases.json]: yeah, it makes sense but only in M:M relations for now. base.json fills out data for 1:M scenario atm, so `one-to-many-extended-cases.json` - doesn't have any use case at the moment (maybe there will be needed more data and then that file could be used to kind of extend the base with more records).

**Pull request merged**